src/Extensions/Hosting.Services/ServiceFabricHostBuilder.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Fabric;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.Omex.Extensions.Hosting.Services
+{
+	/// <summary>
+	/// Wrapper on top of IHostBuilder to propagete proper context type and avoid type registration mistakes
+	/// </summary>
+	public sealed class ServiceFabricHostBuilder<TContext> where TContext : ServiceContext
+	{
+		/// <summary>
+		/// Method should be called only by extensions of this class
+		/// If you are creating service please register dependencies before colling BuildService
```

Comment:
BuildService because this class is only available inside of BuildStateful or BuildStateless methods that would be a bad place for registering dependencies.
Although after making this method internal this comment not that useful.
I've added a bit of clarification.